pygen/pygen_src/riscv_asm_program_gen.py

```diff
@@ -338,7 +338,55 @@ def init_gpr(self):
             self.instr_stream.append(init_string)
 
     def init_floating_point_gpr(self):
-        pass
+        for i in range(rcs.NUM_FLOAT_GPR):
+            # TODO randselect
+            '''
+            vsc.randselect([(1, lambda:self.init_floating_point_gpr_with_spf(i)), 
+        ('RV64D' in rcs.supported_isa, lambda:self.init_floating_point_gpr_with_dpf(i))])
+            '''
+            self.init_floating_point_gpr_with_spf(i)
+        # Initialize rounding mode of FCSR
+        string = "{}fsrmi {}".format(pkg_ins.indent, cfg.fcsr_rm)
+        self.instr_stream.append(string)
+
+    def init_floating_point_gpr_with_spf(self, int_floating_gpr):
+        imm = self.get_rand_spf_value()
+        string = "{}li x{}, {}".format(pkg_ins.indent, cfg.gpr[0].value, hex(imm))
+        self.instr_stream.append(string)
+        string = "{}fmv.w.x f{}, x{}".format(pkg_ins.indent, int_floating_gpr,
+                                             cfg.gpr[0].value)
+        self.instr_stream.append(string)
+
+    def init_floating_point_gpr_with_dpf(self, int_floating_gpr):
+        imm = vsc.bit_t(64)
+        imm = self.get_rand_dpf_value()
+        int_gpr1 = cfg.gpr[0].value
+        int_gpr2 = cfg.gpr[1].value
+
+        string = "{}li x{}, {}".format(pkg_ins.indent, int_gpr1, imm[63:32])
+        self.instr_stream.append(string)
+        # shift to upper 32bits
+        for _ in range(2):
+            string = "{}slli x{}, x{}, 16".format(pkg_ins.indent, int_gpr1, int_gpr1)
+            self.instr_stream.append(string)
+        string = "{}li x{}, {}".format(pkg_ins.indent, int_gpr2, imm[31:0])
+        self.instr_stream.append(string)
+        string = "{}or x{}, x{}, x{}".format(pkg_ins.indent, int_gpr2, int_gpr2, int_gpr1)
+        self.instr_stream.append(string)
+        string = "{}fmv.d.x f{}, x{}".format(pkg_ins.indent, int_floating_gpr, int_gpr2)
+        self.instr_stream.append(string)
```

Comment:
I think it would look better to give the "string" a more meaningful name (e.g., `li_inst`, `or_inst`, etc) and use `extend` instead of `append` to add the three lines altogether at the same time instead of keep repeating the calls to `append` (`self.instr_stream.extend([li_inst, or_instr, ...])`)